components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/constant/SQLConstants.java

```diff
@@ -99,11 +139,29 @@
             "    ATTR_VALUE\n" +
             "  )\n" +
             "VALUES(?, ?, ?, ?)";
+    public static final String INSERT_ATTRIBUTES_MSSQL_OR_DB2 = "MERGE INTO IDN_CONFIG_ATTRIBUTE T USING " +
+            "(VALUES (?, ?, ?, ?)";
+    public static final String INSERT_ATTRIBUTES_ORACLE = "MERGE INTO IDN_CONFIG_ATTRIBUTE T " +
+            "USING (SELECT  ? ID, ? RESOURCE_ID, ? ATTR_KEY, ? ATTR_VALUE from dual ";
     public static final String UPDATE_ATTRIBUTES_H2 = "MERGE INTO\n" +
             "  IDN_CONFIG_ATTRIBUTE KEY(ID) VALUES(?, ?, ?, ?)";
     public static final String INSERT_OR_UPDATE_ATTRIBUTES_MYSQL = "ON DUPLICATE KEY UPDATE " +
             "RESOURCE_ID = VALUES(RESOURCE_ID), ATTR_KEY = VALUES(ATTR_KEY), ATTR_VALUE = VALUES(ATTR_VALUE)";
+    public static final String INSERT_OR_UPDATE_ATTRIBUTES_POSTGRESQL = "ON CONFLICT(ID) DO UPDATE SET RESOURCE_ID = " +
+            "EXCLUDED.RESOURCE_ID, ATTR_KEY = EXCLUDED.ATTR_KEY, " +
+            "ATTR_VALUE = EXCLUDED.ATTR_VALUE";
+    public static final String INSERT_OR_UPDATE_ATTRIBUTES_MSSQL_OR_DB2 = ") S (ID, RESOURCE_ID, ATTR_KEY, " +
+            "ATTR_VALUE) ON T.ID = S.ID WHEN MATCHED THEN UPDATE SET RESOURCE_ID = S.RESOURCE_ID, " +
+            "ATTR_KEY = S.ATTR_KEY, ATTR_VALUE = S.ATTR_VALUE WHEN NOT MATCHED THEN INSERT (ID, RESOURCE_ID, " +
+            "ATTR_KEY, ATTR_VALUE) VALUES (S.ID, S.RESOURCE_ID, S.ATTR_KEY, S.ATTR_VALUE);";
+    public static final String INSERT_OR_UPDATE_ATTRIBUTES_ORACLE = ") S ON (T.ID = S.ID) " +
+            "WHEN MATCHED THEN UPDATE SET T.RESOURCE_ID = S.RESOURCE_ID, T.ATTR_KEY = S.ATTR_KEY, " +
+            "T.ATTR_VALUE = S.ATTR_VALUE\n" +
```

Comment:
remove /n

Reply:
addressed in 5a0504e41b864eb30d7aa555f60249f9b40e79e9